### PR TITLE
[iridium] Iridium dial tool

### PR DESCRIPTION
**Pull request by dewagter**
communicate with Iridium Core 9523 Satellite Transceiver Module

### CODE REVIEW

File: sw/tools/iridium/iridium_link.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+from socket import SOL_SOCKET, SO_BROADCAST
+import serial
+import pygtk
+import gtk
+import os
+pygtk.require('2.0')
+
+
+
+DEFAULT_DEVICE = "/dev/paparazzi/iridium"
```
Comment (gautierhattenberger) [inline sw/tools/iridium/iridium_link.py]: do you have a udev rule for this or is it only an example ?

File: sw/tools/iridium/iridium_link.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python
+from twisted.internet import gtk2reactor
```
Comment (gautierhattenberger) [inline sw/tools/iridium/iridium_link.py]: license header is missing

File: sw/tools/iridium/iridium_link.py [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+DEFAULT_DEVICE = "/dev/paparazzi/iridium"
+DEFAULT_BAUD = "19200"
+DEFAULT_NUMBER = "00881111111111"
+UDP_IN = 4243
+UDP_OUT = 4242
+
+# IridiumCtrl interface
+class IridiumCtrl:
+
+  def open(self, widget, data=None):
```
Comment (gautierhattenberger) [inline sw/tools/iridium/iridium_link.py]: if I'm not wrong, this is not the recommended indentation for python code

Comment (flixr) [inline sw/tools/iridium/iridium_link.py]: please adhere to pep8 (4 spaces indentation, no tabs)

**Comment by flixr** [comment]
Can you please also add a Readme.md that describes the functionality and lists the dependencies (twisted, iridium_protocol, ..)

**Comment by dewagter** [comment]
All iridium stuff was placed in a separate repository: https://github.com/tudelft/iridium/

Is it an option to submodule it and only download when needed?

**Comment by flixr** [comment]
What do you mean by "download when needed"? As soon as someone does a `git submodule update` it will download the submodule, regardless of what rules we add in our makefiles...
Also I don't think that the whole repo is needed as a submodule (including hw files, etc).

It seems that only `iridium_protocol.py` is needed, why not add that directly here?
Or do you expect lots of changes there and it would be annoying to keep in sync?

**Comment by flixr** [comment]
sorry, missed that `iridium_protocol.py` is already added here...

**Comment by flixr** [comment]
So, fixing python indentation, using PAPARAZZI_SRC instead of HOME for the icon and adding a Readme.md and requirements.txt would IMHO make this complete/mergeable...

File: sw/tools/iridium/iridium_link.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    self.cb_recv_data = cb_recv_data
+
+  def startProtocol(self):
+    self.transport.socket.setsockopt(SOL_SOCKET, SO_BROADCAST, True)
+
+  def sendData(self, data):
+    self.transport.write(data, ('127.255.255.255', self.out_port))
+
+  def datagramReceived(self, datagram, addr):
+    if self.cb_recv_data != None:
```
Comment (flixr) [inline sw/tools/iridium/iridium_link.py]: Please not that checking for None should not be done with comparison operators, but with `is` and `is not`.
See https://www.python.org/dev/peps/pep-0008/#programming-recommendations
http://stackoverflow.com/questions/14247373/python-none-comparison-should-i-use-is-or

**Comment by dewagter** [comment]
IMHO all comments were now processed

**Comment by flixr** [comment]
Can you please also replace all the other cases of `foo != None` with `foo is not None`?

**Comment by dewagter** [comment]
Sure!

-Christophe

**Comment by gautierhattenberger** [comment]
can you merge or rebase with master and fix conflict, beside that it's all good

**Comment by dewagter** [comment]
Rebased / Fixed.

**Pull request merged**